lib/cli/cli.js

```diff
@@ -199,19 +193,10 @@ class CLI {
           }
         });
       });
-
-      onCommandInterrupt = () =>
-        Promise.resolve(command.onInterrupt())
-          .then(() => runPromise);
-
-      return runPromise;
-    })
-    .finally(() => {
-      if (shutdownOnExit) {
-        shutdownOnExit();
-      }
-    })
-    .catch(this.logError.bind(this));
+    }).finally(() => {
+      onProcessInterrupt.teardown();
```

Comment:
What is the reason for additional `teardown` here? https://github.com/ember-cli/ember-cli/pull/7019/files#diff-16a41bfbe414387d773df099f941217eR162 should do all the necessary cleanup?